src/graph/generate-chart-data.ts

```diff
@@ -57,11 +57,15 @@ export const generateChartData = (datasets: Array<IDataset | null | undefined>)
     if (!dataset) {
       return;
     }
-    const xPropIdx = dataset.properties.indexOf(dataset.xAxisProp);
-    if (xPropIdx < 0) {
-      throw new Error("Incorrect dataset.xAxisProp value");
-    }
-    const labels = dataset.rows.map(row => row[xPropIdx] || "");
+    const xPropIdx = dataset.xAxisProp ? dataset.properties.indexOf(dataset.xAxisProp) : -1;
+    const labels: Label[] = dataset.rows.map((row, rowIdx) => {
+      if (xPropIdx === -1) {
+        // If x property isn't defined, use row index, but starting from 1.
+        return rowIdx + 1;
+      }
+      const val = row[xPropIdx];
+      return val !== null ? val : "";
```

Comment:
Assuming that `undefined` is not a valid value either, I would use `!= null` here to catch `undefined` as well.

Reply:
`undefined` is not allowed by the interface (because of possible JSON serialization along the way), so it should never show up here.